common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDB.java

```diff
@@ -355,26 +355,22 @@ public void close() throws IOException {
     }
   }
 
-  /**
-   * Closes the given iterator if the DB is still open. Trying to close a JNI RocksDB handle
-   * with a closed DB can cause JVM crashes, so this ensures that situation does not happen.
-   */
-  void closeIterator(RocksDBIterator<?> it) throws IOException {
-    notifyIteratorClosed(it);
-    synchronized (this._db) {
-      org.rocksdb.RocksDB _db = this._db.get();
-      if (_db != null) {
-        it.close();
-      }
-    }
+  AtomicReference<org.rocksdb.RocksDB> getRocksDB() {
+    return _db;
   }
 
-  /**
-   * Remove iterator from iterator tracker. `RocksDBIterator` calls it to notify
-   * iterator is closed.
-   */
-  void notifyIteratorClosed(RocksDBIterator<?> it) {
-    iteratorTracker.removeIf(ref -> it.equals(ref.get()));
+  ConcurrentLinkedQueue<Reference<RocksDBIterator<?>>> getIteratorTracker() {
+    return iteratorTracker;
+  }
+
+  @VisibleForTesting
```

Comment:
hmm.. this method can be moved to RocksDBSuite.java as a private method due to `RocksDB`  already has `getIteratorTracker(`

Reply:
done